pallets/liquidity-proxy/src/lib.rs

```diff
@@ -2441,6 +2441,46 @@ pub mod pallet {
             ADARCommissionRatio::<T>::put(commission_ratio);
             Ok(().into())
         }
+
+        /// Extrinsic which is enable XORless transfers.
+        /// Internally it's swaps `asset_id` to `desired_xor_amount` of `XOR` and transfers remaining amount of `asset_id` to `receiver`.
+        /// Client apps should specify the XOR amount which should be paid as a fee in `desired_xor_amount` parameter.
+        /// If sender will not have enough XOR to pay fees after execution, transaction will be rejected.
+        /// This extrinsic is done as temporary solution for XORless transfers, in future it would be removed
+        /// and logic for XORless extrinsics should be moved to xor-fee pallet.
+        #[pallet::call_index(6)]
+        #[pallet::weight(Pallet::<T>::swap_weight(dex_id, asset_id, &common::XOR.into(), SwapVariant::WithDesiredOutput).saturating_add(<T as assets::Config>::WeightInfo::transfer()))]
+        pub fn xorless_transfer(
+            origin: OriginFor<T>,
+            dex_id: T::DEXId,
+            asset_id: T::AssetId,
+            receiver: T::AccountId,
+            amount: Balance,
+            desired_xor_amount: Balance,
+            max_amount_in: Balance,
+            selected_source_types: Vec<LiquiditySourceType>,
+            filter_mode: FilterMode,
+        ) -> DispatchResultWithPostInfo {
+            let sender = ensure_signed(origin)?;
+            let mut weight = Self::inner_swap(
```

Comment:
Probably we can check the XOR balance of `sender` before swap.
Is there a chance that `sender` already has enough XOR for fee?

Reply:
Done in `xor_fee_impls`. Extrinsics don't know anything about fee amount, so user/client app should specify `desired_xor_amount` to 0 if user already have enough XOR for fees. Added check to skip swap if `desired_xor_amount` or `max_amount_in` is 0